include/xtensor/xchunked_array.hpp

```diff
@@ -0,0 +1,91 @@
+#include <vector>
+#include <array>
+#include "xarray.hpp"
+
+template <class Chunk_type>
+class xchunked_array
+{
+    public:
+
+        using const_reference = typename Chunk_type::const_reference;
+
+        template <class... Args>
+        inline const_reference operator()(Args... args) const
+        {
+            return access_impl(args...);
+        }
+
+        xchunked_array(std::vector<size_t> shape, std::vector<size_t> chunks):
+            m_shape(shape),
+            m_chunk_shape(chunks)
+        {
+            std::vector<size_t> shape_chunk;
+            size_t di = 0;
+            std::cout << "Creating chunked array with shape";
+            for (auto s: shape)
+            {
+                std::cout << " " << s;
+                size_t chunk_nb = s / chunks[di];
+                if (s % chunks[di] > 0)
+                    chunk_nb += 1;  // edge chunk
+                shape_chunk.push_back(chunk_nb);
+                di++;
+            }
+            std::cout << ", chunk shape";
+            for (auto s: chunks)
+                std::cout << " " << s;
+            std::cout << std::endl;
+            m_chunks.resize(shape_chunk);
+        }
+
+    private:
+
+        template <class Idx, class Arg>
+        size_t get_index_of_chunk_in_dimension(Idx idx, Arg arg) const
+        {
+            size_t index_of_chunk = arg / m_chunk_shape[idx];
+            return index_of_chunk;
+        }
+
+        template <class Idx, class Arg>
+        size_t get_index_in_chunk_in_dimension(Idx idx, Arg arg) const
+        {
+            std::cout << " " << arg;
+            size_t index_of_chunk = get_index_of_chunk_in_dimension(idx, arg);
+            size_t index_in_chunk = arg - index_of_chunk * m_chunk_shape[idx];
+            return index_in_chunk;
+        }
```

Comment:
These to methods should be merged to avoid calling `get_index_of_chunk_in_dimension` twice

Reply:
The problem if we do that and return both results in a tuple, is that we end up with an array of tuples instead of a tuple of arrays, which we will have to "unzip" into a tuple of arrays in order to pass each array to the `.element()` method, which is not optimal either. Or do you see another way?

Reply:
I agree that it is not optimal, but copying is still faster than performing divisions. Besides, `std::array` are allocated on the stack so this might not hit performance that much.

Reply:
:+1: